master/internal/api_workspace.go

```diff
@@ -358,8 +391,16 @@ func (a *apiServer) PostWorkspace(
 		}
 	}
 
-	_, err = tx.NewInsert().Model(w).Exec(ctx)
+	if err = workspace.AuthZProvider.Get().
+		CanModifyWorkspaceNamespaceBindings(ctx, *curUser); err != nil {
+		return nil, status.Error(codes.PermissionDenied, err.Error())
+	}
 
+	if err := validatePostWorkspaceRequest(req); err != nil {
+		return nil, err
+	}
+
+	err = tx.NewInsert().Model(w).Scan(ctx, w)
```

Comment:
nit: since you're editing this function anyway, put the db transactions into a `db.Bun().RunInTx(...)` -- that will replace lines 356-364